plugins/chunter-resources/src/components/ChannelTypingInfo.svelte

```diff
@@ -15,47 +15,57 @@
 <script lang="ts">
   import { onMount } from 'svelte'
   import chunter from '@hcengineering/chunter'
-  import { getName, getCurrentEmployee } from '@hcengineering/contact'
+  import { getName, getCurrentEmployee, Person } from '@hcengineering/contact'
   import { getPersonsByPersonRefs } from '@hcengineering/contact-resources'
   import { getClient } from '@hcengineering/presentation'
   import { Label } from '@hcengineering/ui'
-  import { PresenceTyping } from '../types'
+  import { subscribeTyping } from '@hcengineering/presence-resources'
+  import { type Ref } from '@hcengineering/core'
 
-  export let typingInfo: PresenceTyping[] = []
-
-  const typingDelay = 2000
   const maxTypingPersons = 3
   const me = getCurrentEmployee()
   const hierarchy = getClient().getHierarchy()
 
   let typingPersonsLabel: string = ''
-  let typingPersonsCount = 0
+  let typingPersonsCount = 0  
   let moreCount: number = 0
 
-  $: void updateTypingPersons(typingInfo)
+  $: void updateTypingPersons()
 
-  async function updateTypingPersons (typingInfo: PresenceTyping[]): Promise<void> {
-    const now = Date.now()
-    const personIds = new Set(
-      typingInfo.filter((info) => info.person !== me && now - info.lastTyping < typingDelay).map((info) => info.person)
-    )
-    const persons = await getPersonsByPersonRefs(Array.from(personIds))
-    const names = Array.from(persons.values())
-      .map((person) => getName(hierarchy, person))
-      .sort((name1, name2) => name1.localeCompare(name2))
+  export let typingInfo: Ref<Person>[] = []
 
+  async function updateTypingPersons (): Promise<void> {
+    const persons = await getPersonsByPersonRefs(typingInfo)
+    const names = Array.from(persons.values())
+        .map((person) => getName(hierarchy, person))
+        .sort((name1, name2) => name1.localeCompare(name2))
     typingPersonsCount = names.length
     typingPersonsLabel = names.slice(0, maxTypingPersons).join(', ')
     moreCount = Math.max(names.length - maxTypingPersons, 0)
   }
 
-  onMount(() => {
-    const interval = setInterval(() => {
-      void updateTypingPersons(typingInfo)
-    }, typingDelay)
-    return () => {
-      clearInterval(interval)
+  function handleTypingInfo(msg: any, key: String, index: number): void {
+    try {
+      const id: Ref<Person> = msg.key.split('/').pop() as Ref<Person>
```

Comment:
How do you filter by channel? This code will show that someone is typing even when they type in different channel (even private).